### PR TITLE
Gitty Solution

### PR DESCRIPTION
**Pull request by haakjes**

### CODE REVIEW

File: gizmo.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+
+def generate_fibonacci_sequence(n):
+    a = 0
+    b = 1
+    for _ in range(n):
+        yield a
+        c = a + b
+        a = b
+        b = c
```
Comment (wmvanvliet) [inline gizmo.py]: Yielding `a` is elegant!

File: gizmo.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    def speak(self):
+        print(self.name)
+
+
+def hello(name, country="Finland"):
+    print(f"Hello {name}, how are things in {country}?")
+
+
+def spell(word):
+    print(".".join(word))
```
Comment (wmvanvliet) [inline gizmo.py]: I never realized you wouldn't need a `word.split()`, you can just `join()` straight away.

Anyway, I've added a unit test that looks for `join` and `split` in the code of your function, no more cheating ;-)

File: gizmo.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        When this parameter is set to an integer number, then the multiplication table that is returned by the function will have all multiples of the given number set to zero.
+
+    Returns
+    -------
+    multiplication_table : ndarray
+        Multiplication table
+    """
+    array = np.arange(1, 13)
+    if zero_out_multiples:
+        array[array % zero_out_multiples == 0] = 0
```
Comment (wmvanvliet) [inline gizmo.py]: I like that you do this before the `np.outer()`. Efficiency!

**Review by wmvanvliet — APPROVED** [review]
Nice work!

File: gizmo.py (hunk 3 above)
Comment (haakjes) [reply gizmo.py]: If only it would not introduce a bug where zero_out_multiples = 6 and 3*4=12 instead of 0

File: gizmo.py (hunk 2 above)
Comment (haakjes) [reply gizmo.py]: Fixed the "Cheating" :)

**Review by wmvanvliet — APPROVED** [review]

**Pull request closed without merging**